src/persist-client/examples/maelstrom/txn.rs

```diff
@@ -149,7 +149,9 @@ impl Transactor {
             // NB: We do the CaS even if writes is empty, so that read-only txns
             // are also linearizable.
             let write_ts = self.read_ts + 1;
-            let updates = writes.iter().map(|(k, v, diff)| ((k, v), &write_ts, diff));
+            let updates = writes
```

Comment:
I don't like that this change is necessary, even though the public API didn't change. (It's bleeding through in the Future type of the function)

When this is not changed, we get:
```
error: implementation of `FnOnce` is not general enough
   --> src/persist-client/examples/maelstrom/txn.rs:448:66
    |
448 |       async fn eval(&self, handle: Handle, src: NodeId, req: Body) {
    |  __________________________________________________________________^
449 | |         match req {
450 | |             Body::ReqTxn { msg_id, txn } => {
451 | |                 let in_reply_to = msg_id;
...   |
469 | |         }
470 | |     }
    | |_____^ implementation of `FnOnce` is not general enough
    |
    = note: closure with signature `fn(&'0 (MaelstromKey, MaelstromVal, i64)) -> ((&MaelstromKey, &MaelstromVal), &u64, &i64)` must implement `FnOnce<(&(MaelstromKey, MaelstromVal, i64),)>`, for any lifetime `'0`...
    = note: ...but it actually implements `FnOnce<(&(MaelstromKey, MaelstromVal, i64),)>`
```

And I haven't yet figured out why that is. It must have something to do with how the internal code structure behind the function call is different now.

Reply:
I'm quite sure it's because the borrowed references are now held across await points (because all of `create_builder`, `add`, and `finish` are `async`), which makes the lifetime bleed into the Future signature of `compare_and_append`. Before, they were borrowed in `encode_batch()`, but never across an await point.

Reply:
ah okay. well then this probably would have happened when we did the "bounded memory usage" stuff because that would also introduce an await point (when the columnar batch fills up and is flushed out into a blob)